### PR TITLE
LPS-174179 The user should not be allowed to mark a picklist field as state on System Objects

### PR DESCRIPTION
**Pull request by diogomelo09**
[LPS-174179](https://issues.liferay.com/browse/LPS-174179)

### CODE REVIEW

**Comment by diogomelo09** [comment]
ci:reevaluate:1327401_8156

**Comment by diogomelo09** [comment]
ci:reevaluate:1357401_3559

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx [hunk 1]
```diff
@@ -437,7 +437,7 @@ export default function ObjectFieldFormBase({
 				{values.businessType === 'Picklist' &&
 					validListTypeDefinitionId && (
 						<ClayToggle
-							disabled={disabled}
+							disabled={true}
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx:440]: Hi @diogomelo09, if you simply set the disable property as true, the user will never be able to mark any picklist fields as state. The issue described in the ticket is just related with system objects. So you should set the property as `disabled={disabled || objectDefinition?.system}`.

**Review by naraandrade — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx (hunk 1 above)
Comment (diogomelo09) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx]: Just did it!! :peepo-clap:

**Review by naraandrade — APPROVED** [review]

**Comment by naraandrade** [comment]
ci:test:object

**Comment by naraandrade** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
ci:forward

**Comment by naraandrade** [comment]
Hi @diogomelo09, I've checked the failures and they don't seem related to your changes. You can send this PR manually to BChan. Please, link the new PR here before closing this one. Thanks :)

**Comment by diogomelo09** [comment]
Resent here:[#130934](https://github.com/brianchandotcom/liferay-portal/pull/130934)

**Pull request closed without merging**